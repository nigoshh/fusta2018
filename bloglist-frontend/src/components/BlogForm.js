import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { newBlog } from '../reducers/blogReducer'

export class BlogForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      author: '',
      title: '',
      url: ''
    }
  }

  addBlog = async (event) => {
    event.preventDefault()
    const { author, title, url } = this.state
    this.props.newBlog(
      { author, title, url }, this.cleanForm, this.props.history
    )
  }

  cleanForm = () => this.setState({ author: '', title: '', url: '' })

  handleChange = (event) =>
    this.setState({ [event.target.name]: event.target.value })

  render() {
    const { author, title, url } = this.state
    return (
      <div>
        <h3>add blog</h3>
        <form onSubmit={this.addBlog}>
          <div>
            title: <input
              name="title"
              value={title}
              onChange={this.handleChange}
            />
          </div>
          <div>
            author: <input
              name="author"
              value={author}
              onChange={this.handleChange}
            />
          </div>
          <div>
            url: <input
              name="url"
              value={url}
              onChange={this.handleChange}
            />
          </div>
          <div>
            <button type="submit">create</button>
          </div>
        </form>
      </div>
    )
  }
}

BlogForm.propTypes = {
  newBlog: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
}

export default connect(null, { newBlog })(BlogForm)
