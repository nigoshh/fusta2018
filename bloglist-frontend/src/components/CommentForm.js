import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { newComment } from '../reducers/blogReducer'

export class CommentForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      comment: ''
    }
  }

  addComment = event => {
    event.preventDefault()
    this.props.newComment(
      this.props.blogId, this.state.comment, this.cleanForm, this.props.history
    )
  }

  cleanForm = () => this.setState({ comment: '' })

  handleChange = event =>
    this.setState({ [event.target.name]: event.target.value })

  render() {
    return (
      <form onSubmit={this.addComment}>
        <input
          type="text"
          name="comment"
          value={this.state.comment}
          onChange={this.handleChange}
        />
        <button type="submit">add comment</button>
      </form>
    )
  }
}

CommentForm.propTypes = {
  blogId: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired,
  newComment: PropTypes.func.isRequired
}

export default connect(null, { newComment })(CommentForm)
