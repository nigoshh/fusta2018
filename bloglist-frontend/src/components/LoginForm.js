import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { login } from '../reducers/userReducer'

export class LoginForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: ''
    }
  }

  cleanForm = () => this.setState({ username: '', password: '' })

  handleChange = event =>
    this.setState({ [event.target.name]: event.target.value })

  login = event => {
    event.preventDefault()
    const { username, password } = this.state
    const { history, url } = this.props
    this.props.login({ username, password }, this.cleanForm, history, url)
  }

  render() {
    const { username, password } = this.state
    return (
      <div>
        <h2>log in</h2>
        <form onSubmit={this.login}>
          <div>
            username: <input
              type="text"
              name="username"
              value={username}
              onChange={this.handleChange}
              ref={this.props.focusRef}
            />
          </div>
          <div>
            password: <input
              type="password"
              name="password"
              value={password}
              onChange={this.handleChange}
            />
          </div>
          <button type="submit">log in</button>
        </form>
      </div>
    )
  }
}

LoginForm.propTypes = {
  focusRef: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  login: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired
}

export default connect(null, { login })(LoginForm)
