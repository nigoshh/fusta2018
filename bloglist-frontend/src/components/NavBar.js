import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const NavBar = ({ loginInfo, logoutButton }) => {
  const baseStyle = { padding: '15px 10px' }
  const style = { ...baseStyle, backgroundColor: '#33cccc' }
  const active = {
    ...baseStyle,
    backgroundColor: '#006699',
    color: 'white',
    fontStyle: 'italic'
  }
  return (
    <div style={style}>
      <NavLink exact to="/blogs" style={style} activeStyle={active}>
        blogs
      </NavLink>&nbsp;
      <NavLink exact to="/users" style={style} activeStyle={active}>
        users
      </NavLink>&nbsp;
      {loginInfo}
      {logoutButton}
    </div>
  )
}

NavBar.propTypes = {
  loginInfo: PropTypes.object.isRequired,
  logoutButton: PropTypes.object.isRequired
}

export default NavBar
