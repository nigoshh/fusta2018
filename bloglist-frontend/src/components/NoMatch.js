import React from 'react'

const NoMatch = () =>
  <h2>{'404 this resource doesn\'t exist'}</h2>

export default NoMatch
