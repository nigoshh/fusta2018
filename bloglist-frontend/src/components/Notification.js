import React from 'react'
import PropTypes from 'prop-types'

const Notification = ({ message, className }) => {
  if (message)
    return <div className={className}>{message}</div>
  return null
}

Notification.propTypes = {
  message: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired
}

export default Notification
