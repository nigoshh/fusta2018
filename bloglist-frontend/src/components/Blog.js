import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { likeBlog, removeBlog } from '../reducers/blogReducer'
import CommentForm from './CommentForm'
import NoMatch from './NoMatch'

export const Blog = ({ blog, history, like, loggedInUser, remove }) => {

  if (!blog)
    return <NoMatch />

  const { author, comments, id, likes, title, url, user } = blog
  const deleteButton =
    <button onClick={() => remove(blog, history)}>delete</button>
  const likeButton = <button onClick={() => like(blog, history)}>like</button>
  const superUser =
    <span style={{ textDecoration: 'line-through' }}>secret user</span>
  const commentList = () =>
    <ul>
      {comments.map(c => <li key={c}>{c}</li>)}
    </ul>

  return (
    <div>
      <h2><span style={f('italic')}>{title}</span> â€“ {author}</h2>
      <div className="box">
        <a href={url}>{url}</a><br/>
        {likes} likes {likeButton}<br/>
        added by {user ? user.name : superUser}<br/>
        {(!user || user.id === loggedInUser.id) ? deleteButton : ''}
        <h3>comments</h3>
        {comments ? commentList() : null}
        <CommentForm blogId={id} history={history} />
      </div>
    </div>
  )
}

export const f = fontStyle => ({ fontStyle })

Blog.propTypes = {
  blog: PropTypes.object,
  history: PropTypes.object.isRequired,
  like: PropTypes.func.isRequired,
  loggedInUser: PropTypes.object.isRequired,
  remove: PropTypes.func.isRequired
}

const mapStateToProps = ({ blogs, loggedInUser }, { history, id }) =>
  ({ blog: blogs.find(b => b.id === id), history, loggedInUser })

const mapDispatchToProps = { like: likeBlog, remove: removeBlog }

export default connect(mapStateToProps, mapDispatchToProps)(Blog)
