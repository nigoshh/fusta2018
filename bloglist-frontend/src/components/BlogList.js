import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { f } from './Blog'

export const BlogList = ({ blogs }) =>
  blogs.map(b =>
    <div className="box" key={b.id}>
      <Link to={`/blogs/${b.id}`}>
        <span style={f('italic')}>{b.title}</span> â€“ {b.author}
      </Link>
    </div>
  )

BlogList.propTypes = {
  blogs: PropTypes.array.isRequired
}

export default connect(({ blogs }) => ({ blogs }))(BlogList)
