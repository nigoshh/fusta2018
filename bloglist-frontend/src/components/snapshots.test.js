import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { blogs, loggedInUser, users } from '../test_utils/test_helper'
import { BrowserRouter as Router } from 'react-router-dom'
import { Blog } from './Blog'
import { BlogForm } from './BlogForm'
import { BlogList } from './BlogList'
import { CommentForm } from './CommentForm'
import { LoginForm } from './LoginForm'
import NavBar from './NavBar'
import NoMatch from './NoMatch'
import Notification from './Notification'
import SimpleBlog from './SimpleBlog'
import Togglable from './Togglable'
import { User } from './User'
import { UserList } from './UserList'

const fc = <div/>
const ff = () => {}
const fo = {}
const fs = 's'

it('renders <Blog/> correctly', () => {
  blogs.forEach(b => {
    const component = shallow(
      <Blog
        blog={b} history={fo} like={ff} loggedInUser={loggedInUser} remove={ff}
      />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})

it('renders <BlogForm/> correctly', () => {
  const tree = renderer
    .create(<BlogForm newBlog={ff} history={fo} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <BlogList/> correctly', () => {
  const tree = renderer
    .create(<Router><BlogList blogs={blogs} /></Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <CommentForm/> correctly', () => {
  const tree = renderer
    .create(<CommentForm blogId={blogs[0].id} history={fo} newComment={ff} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <LoginForm/> correctly', () => {
  const tree = renderer
    .create(<LoginForm focusRef={ff} history={fo} login={ff} url={fs} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <NavBar/> correctly', () => {
  const tree = renderer
    .create(<Router><NavBar loginInfo={fc} logoutButton={fc} /></Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <NoMatch/> correctly', () => {
  const tree = renderer
    .create(<NoMatch/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <Notification/> correctly', () => {
  ['error', 'message'].forEach(s => {
    const tree = renderer
      .create(<Notification message={s} className={s} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it('renders <SimpleBlog/> correctly', () => {
  const tree = renderer
    .create(<SimpleBlog blog={blogs[0]} onClick={ff} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <Togglable/> correctly', () => {
  const tree = renderer
    .create(<Togglable buttonLabel={fs}>{fc}</Togglable>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <User/> correctly', () => {
  const tree = renderer
    .create(<User user={users[0]} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders <UserList/> correctly', () => {
  const tree = renderer
    .create(<Router><UserList users={users} /></Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
