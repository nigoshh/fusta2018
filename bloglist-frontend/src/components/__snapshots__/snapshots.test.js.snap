// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Blog/> correctly 1`] = `
<div>
  <h2>
    <span
      style={
        Object {
          "fontStyle": "italic",
        }
      }
    >
      Paperopoli
    </span>
     – 
    Zio
  </h2>
  <div
    className="box"
  >
    <a
      href="httpke"
    >
      httpke
    </a>
    <br />
    7
     likes 
    <button
      onClick={[Function]}
    >
      like
    </button>
    <br />
    added by 
    Papera
    <br />
    <h3>
      comments
    </h3>
    <ul>
      <li
        key="yo"
      >
        yo
      </li>
      <li
        key="cheers"
      >
        cheers
      </li>
      <li
        key="mate"
      >
        mate
      </li>
    </ul>
    <Connect(CommentForm)
      blogId="5b2b843c78c7482a2031d9eb"
      history={Object {}}
    />
  </div>
</div>
`;

exports[`renders <Blog/> correctly 2`] = `
<div>
  <h2>
    <span
      style={
        Object {
          "fontStyle": "italic",
        }
      }
    >
      Cash
    </span>
     – 
    Spitty
  </h2>
  <div
    className="box"
  >
    <a
      href="httpmah"
    >
      httpmah
    </a>
    <br />
    23
     likes 
    <button
      onClick={[Function]}
    >
      like
    </button>
    <br />
    added by 
    Lia
    <br />
    <button
      onClick={[Function]}
    >
      delete
    </button>
    <h3>
      comments
    </h3>
    <ul>
      <li
        key="nice stuff"
      >
        nice stuff
      </li>
    </ul>
    <Connect(CommentForm)
      blogId="5b2b849178c7482a2031d9ed"
      history={Object {}}
    />
  </div>
</div>
`;

exports[`renders <Blog/> correctly 3`] = `
<div>
  <h2>
    <span
      style={
        Object {
          "fontStyle": "italic",
        }
      }
    >
      Jam
    </span>
     – 
    Jazz
  </h2>
  <div
    className="box"
  >
    <a
      href="httpb"
    >
      httpb
    </a>
    <br />
    0
     likes 
    <button
      onClick={[Function]}
    >
      like
    </button>
    <br />
    added by 
    Lia
    <br />
    <button
      onClick={[Function]}
    >
      delete
    </button>
    <h3>
      comments
    </h3>
    <ul />
    <Connect(CommentForm)
      blogId="5b2bbd95423e6c3af4c4aaa4"
      history={Object {}}
    />
  </div>
</div>
`;

exports[`renders <BlogForm/> correctly 1`] = `
<div>
  <h3>
    add blog
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div>
      title: 
      <input
        name="title"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      author: 
      <input
        name="author"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      url: 
      <input
        name="url"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      <button
        type="submit"
      >
        create
      </button>
    </div>
  </form>
</div>
`;

exports[`renders <BlogList/> correctly 1`] = `
Array [
  <div
    className="box"
>
    <a
        href="/blogs/5b2b843c78c7482a2031d9eb"
        onClick={[Function]}
    >
        <span
            style={
                Object {
                    "fontStyle": "italic",
                  }
            }
        >
            Paperopoli
        </span>
         – 
        Zio
    </a>
</div>,
  <div
    className="box"
>
    <a
        href="/blogs/5b2b849178c7482a2031d9ed"
        onClick={[Function]}
    >
        <span
            style={
                Object {
                    "fontStyle": "italic",
                  }
            }
        >
            Cash
        </span>
         – 
        Spitty
    </a>
</div>,
  <div
    className="box"
>
    <a
        href="/blogs/5b2bbd95423e6c3af4c4aaa4"
        onClick={[Function]}
    >
        <span
            style={
                Object {
                    "fontStyle": "italic",
                  }
            }
        >
            Jam
        </span>
         – 
        Jazz
    </a>
</div>,
]
`;

exports[`renders <CommentForm/> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="comment"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    add comment
  </button>
</form>
`;

exports[`renders <LoginForm/> correctly 1`] = `
<div>
  <h2>
    log in
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div>
      username: 
      <input
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      password: 
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      log in
    </button>
  </form>
</div>
`;

exports[`renders <NavBar/> correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#33cccc",
      "padding": "15px 10px",
    }
  }
>
  <a
    aria-current={null}
    className={undefined}
    href="/blogs"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#33cccc",
        "padding": "15px 10px",
      }
    }
  >
    blogs
  </a>
   
  <a
    aria-current={null}
    className={undefined}
    href="/users"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#33cccc",
        "padding": "15px 10px",
      }
    }
  >
    users
  </a>
   
  <div />
  <div />
</div>
`;

exports[`renders <NoMatch/> correctly 1`] = `
<h2>
  404 this resource doesn't exist
</h2>
`;

exports[`renders <Notification/> correctly 1`] = `
<div
  className="error"
>
  error
</div>
`;

exports[`renders <Notification/> correctly 2`] = `
<div
  className="message"
>
  message
</div>
`;

exports[`renders <SimpleBlog/> correctly 1`] = `
<div>
  <div
    className="sBlogInfo"
  >
    Paperopoli
     
    Zio
  </div>
  <div
    className="sBlogLikes"
  >
    blog has 
    7
     likes
    <button
      onClick={[Function]}
    >
      like
    </button>
  </div>
</div>
`;

exports[`renders <Togglable/> correctly 1`] = `
<div>
  <div
    style={
      Object {
        "display": "",
      }
    }
  >
    <button
      name="s"
      onClick={[Function]}
    >
      s
    </button>
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div />
    <button
      onClick={[Function]}
    >
      hide 
      "s"
    </button>
  </div>
</div>
`;

exports[`renders <User/> correctly 1`] = `
<div>
  <h2>
    Lia
  </h2>
  <div
    className="box"
  >
    <h3>
      added blogs
    </h3>
    <ul>
      <li>
        <span
          style={
            Object {
              "fontStyle": "italic",
            }
          }
        >
          Cash
        </span>
         – 
        Spitty
      </li>
      <li>
        <span
          style={
            Object {
              "fontStyle": "italic",
            }
          }
        >
          Jam
        </span>
         – 
        Jazz
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders <UserList/> correctly 1`] = `
<div>
  <h2>
    users
  </h2>
  <table>
    <thead>
      <tr>
        <th>
          user
        </th>
        <th>
          blogs added
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/users/5b2acb1f140b1521cc5b5e36"
            onClick={[Function]}
          >
            Lia
          </a>
        </td>
        <td>
          2
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/users/5b2acb1f140b1521cc5b5e34"
            onClick={[Function]}
          >
            Papera
          </a>
        </td>
        <td>
          1
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/users/5b2acb1f140b1521cc5b5e37"
            onClick={[Function]}
          >
            Rino
          </a>
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
